File: 80.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return len(nums)

        p2 = 2
        for i in range(2, len(nums)):
            if nums[i] != nums[p2 - 2]:
                nums[p2] = nums[i]
                p2 += 1

        return p2


sol = Solution()
nums = [1,1,1,2,2,3]
print(sol.removeDuplicates(nums))

File: 80.cpp
//
// Created by DuLi on 2023/9/4.
//
#include <iostream>
using std::vector;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 3)
            return nums.size();

        int p = 2;
        for (int i = 2; i < nums.size(); ++i) {
            if (nums[i] != nums[p-2]){
                nums[p] = nums[i];
                p++;
            }
        }
        return p;
    }
};